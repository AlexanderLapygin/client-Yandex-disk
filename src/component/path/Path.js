import React, {Component} from 'react';
import {withRouter} from 'react-router-dom';
import {connect} from 'react-redux'
import {getToken, getPath} from "./../../selectors/selectors";

class Path extends Component {
//меняем путь роута при нажатии на ссылку в хлебныхкрошках
    handleClickPath() {
        if (this.props.pathEl !== this.props.currentPath[this.props.currentPath.length - 1]) {
            let poz = this.props.currentPath.indexOf(this.props.pathEl);
            let nexPath = '/' + this.props.currentPath.slice(1, poz + 1).join('/');
            this.props.history.push(nexPath);
        }
    }

    render() {
        return (
            <li className="breadcrumb-item " aria-current="page">
                <button className='btn btn-link pl-0 pr-0'
                        onClick={() => this.handleClickPath()}>{this.props.pathEl}</button>
            </li>
        );
    }
}

const mapStateToProps = state => {
    return ({
        token: getToken(state),
        currentPath: getPath(state)
    })
};

export default withRouter(connect(mapStateToProps)(Path));
