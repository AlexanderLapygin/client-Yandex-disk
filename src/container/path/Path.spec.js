import React from 'react';
import { shallow } from 'enzyme';
import { Path } from './Path';

describe('Path testing', () => {

// описываем props
  const props = {
    token: '',
    pathEl: 'folder1',
    currentPath: ['/', 'folder1'],
  };

  describe('Path', () => {

    const PathContainer = shallow(<Path  {...props} />);

    it('snapshot', () => {
      expect(PathContainer).toMatchSnapshot();
     });

    it('button text', () => {
      expect(PathContainer.find('.btnPath').text()).toEqual('folder1');
    });

  });
})
;
