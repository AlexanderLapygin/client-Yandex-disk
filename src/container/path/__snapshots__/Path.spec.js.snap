// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Path testing Path snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Path
    currentPath={
      Array [
        "/",
        "folder1",
      ]
    }
    pathEl="folder1"
    token=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-current": "page",
      "children": <button
        className="btn btn-link pl-0 pr-0 btnPath"
        onClick={[Function]}
      >
        folder1
      </button>,
      "className": "breadcrumb-item ",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "folder1",
        "className": "btn btn-link pl-0 pr-0 btnPath",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": "folder1",
      "type": "button",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-current": "page",
        "children": <button
          className="btn btn-link pl-0 pr-0 btnPath"
          onClick={[Function]}
        >
          folder1
        </button>,
        "className": "breadcrumb-item ",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "folder1",
          "className": "btn btn-link pl-0 pr-0 btnPath",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "folder1",
        "type": "button",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
